src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java

```diff
@@ -132,8 +141,10 @@ public boolean accept(File dir, String name) {
 
         LOGGER.finest("Found " + gcLogs.length + " matching files in " + parentDirectory.getAbsolutePath());
         for (File gcLog : gcLogs) {
-            LOGGER.finest("Adding '" + gcLog.getName() + "' file");
-            result.add(new UnfilteredFileContent(GCLOGS_BUNDLE_ROOT + "{0}", new String[]{gcLog.getName()}, gcLog));
+            if (GCLOGS_RETENTION_DAYS <= 0 || (gcLog.lastModified() > (System.currentTimeMillis() - TimeUnit.DAYS.toMillis(GCLOGS_RETENTION_DAYS)))) {
```

Comment:
I'd prefer if this was into a utility method to make it a bit cleaner

Reply:
Done